test/versioned/mongodb/collection-misc.tap.js

```diff
@@ -14,131 +14,94 @@ function verifyAggregateData(t, data) {
+  t.equal(data.deletedCount, 30)
+  verify(null, [`${STATEMENT_PREFIX}/bulkWrite`], ['bulkWrite'], { strict: false })
 })
 
-common.test('count', function countTest(t, collection, verify) {
-  collection.count(function onCount(err, data) {
-    t.error(err)
-    t.equal(data, 30)
-    verify(null, [`${STATEMENT_PREFIX}/count`, 'Callback: onCount'], ['count'])
-  })
+common.test('count', async function countTest(t, collection, verify) {
+  const data = await collection.count()
+  t.equal(data, 30)
+  verify(null, [`${STATEMENT_PREFIX}/count`], ['count'], { strict: false })
 })
 
-common.test('distinct', function distinctTest(t, collection, verify) {
-  collection.distinct('mod10', function done(err, data) {
-    t.error(err)
-    t.same(data.sort(), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
-    verify(null, [`${STATEMENT_PREFIX}/distinct`, 'Callback: done'], ['distinct'])
-  })
+common.test('distinct', async function distinctTest(t, collection, verify) {
+  const data = await collection.distinct('mod10')
+  t.same(data.sort(), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+  verify(null, [`${STATEMENT_PREFIX}/distinct`], ['distinct'], { strict: false })
 })
 
-common.test('drop', function dropTest(t, collection, verify) {
-  collection.drop(function done(err, data) {
-    t.error(err)
-    t.equal(data, true)
-    verify(null, [`${STATEMENT_PREFIX}/drop`, 'Callback: done'], ['drop'])
-  })
+common.test('drop', async function dropTest(t, collection, verify) {
+  const data = await collection.drop()
+  t.equal(data, true)
+  verify(null, [`${STATEMENT_PREFIX}/drop`], ['drop'], { strict: false })
 })
 
-if (semver.satisfies(pkgVersion, '<3')) {
-  common.test('geoNear', function geoNearTest(t, collection, verify) {
-    collection.ensureIndex({ loc: '2d' }, { bucketSize: 1 }, indexed)
+common.test('isCapped', async function isCappedTest(t, collection, verify) {
+  const data = await collection.isCapped()
+  t.notOk(data)
 
-    function indexed(err) {
-      t.error(err)
-      collection.geoNear(20, 20, { maxDistance: 5 }, done)
-    }
+  verify(null, [`${STATEMENT_PREFIX}/isCapped`], ['isCapped'], { strict: false })
+})
 
-    function done(err, data) {
-      t.error(err)
-      t.equal(data.ok, 1)
-      t.equal(data.results.length, 2)
-      t.equal(data.results[0].obj.i, 21)
-      t.equal(data.results[1].obj.i, 17)
-      t.same(data.results[0].obj.loc, [21, 21])
-      t.same(data.results[1].obj.loc, [17, 17])
-      t.equal(data.results[0].dis, 1.4142135623730951)
-      t.equal(data.results[1].dis, 4.242640687119285)
-      verify(
-        null,
-        [
-          `${STATEMENT_PREFIX}/ensureIndex`,
-          'Callback: indexed',
-          `${STATEMENT_PREFIX}/geoNear`,
-          'Callback: done'
-        ],
-        ['ensureIndex', 'geoNear']
-      )
-    }
-  })
-}
+common.test('options', async function optionsTest(t, collection, verify) {
+  const data = await collection.options()
+
+  // Depending on the version of the mongo server this will change.
```

Comment:
Can you expand on this comment? Is it saying that >=5 is the first branch?

Reply:
none of these tests changed. they were just refactored to be async.  Judging by the test sometimes data is returned or not